daemon/graphdriver/overlay2/overlay.go

```diff
@@ -312,23 +322,33 @@ func (d *Driver) Cleanup() error {
 // CreateReadWrite creates a layer that is writable for use as a container
 // file system.
 func (d *Driver) CreateReadWrite(id, parent string, opts *graphdriver.CreateOpts) error {
+	if opts != nil && len(opts.StorageOpt) != 0 && !projectQuotaSupported {
+		return fmt.Errorf("--storage-opt is supported only for overlay over xfs with 'pquota' mount option")
+	}
+
+	if opts == nil {
+		opts = &graphdriver.CreateOpts{}
+	}
+
+	if len(opts.StorageOpt) == 0 {
+		opts.StorageOpt = map[string]string{
+			"size": fmt.Sprintf("%d", d.options.quota.Size),
```

Comment:
I understand the motivation, I just don't understand how this code helps or even does the right thing to prevent default quota from being applied on ReadOnly layers.
in Create(), the default driver.options.quota.Size is checked regardless of whether you manipulate StorageOpt here or not. Am I wrong?